extensions/sql/contract-definition/loader/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/SqlContractDefinitionLoaderTest.java

```diff
@@ -0,0 +1,138 @@
+
+@ExtendWith(EdcExtension.class)
+class SqlContractDefinitionLoaderTest {
+
+    private final Map<String, String> systemProperties = new HashMap<>() {
+        {
+            put(ConfigurationKeys.DATASOURCE_SETTING_URL, "jdbc:h2:mem:test");
+            put(ConfigurationKeys.DATASOURCE_SETTING_DRIVER_CLASS, org.h2.Driver.class.getName());
+            put(ConfigurationKeys.DATASOURCE_SETTING_NAME, ConfigurationKeys.DATASOURCE_NAME);
+        }
+    };
+    private volatile DataSourceRegistry dataSourceRegistry;
+    private volatile TransactionContext transactionContext;
+    private volatile ContractDefinitionLoader contractDefinitionLoader;
+
+    @BeforeEach
+    void setUp(EdcExtension extension) {
+        extension.registerSystemExtension(ConfigurationExtension.class, (ConfigurationExtension) () -> ConfigFactory.fromMap(systemProperties));
+        extension.registerSystemExtension(ServiceExtension.class, new ServiceExtension() {
+            private ServiceExtensionContext context;
+
+            public void initialize(ServiceExtensionContext context) {
+                this.context = context;
+            }
+
+            public void start() {
+                SqlContractDefinitionLoaderTest.this.transactionContext = context.getService(TransactionContext.class);
+                SqlContractDefinitionLoaderTest.this.dataSourceRegistry = context.getService(DataSourceRegistry.class);
+                SqlContractDefinitionLoaderTest.this.contractDefinitionLoader = context.getService(ContractDefinitionLoader.class);
+
+                try (var inputStream = this.getClass().getClassLoader().getResourceAsStream("schema.sql")) {
+                    var schema = new String(Objects.requireNonNull(inputStream).readAllBytes(), StandardCharsets.UTF_8);
+                    try (var connection = dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection()) {
+                        transactionContext.execute(() -> SqlQueryExecutor.executeQuery(connection, schema));
+                    }
+                } catch (IOException | SQLException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+    }
+
+    @AfterEach
+    void tearDown() {
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection()) {
+                executeQuery(connection, String.format("DELETE FROM %s", SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    @DisplayName("Context Loads, tables exist")
+    void contextLoads() throws SQLException {
+        var query = String.format("SELECT 1 FROM %s", SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE);
+        executeQuery(dataSourceRegistry.resolve(ConfigurationKeys.DATASOURCE_NAME).getConnection(), query);
```

Comment:
this test is missing an assertion. Tests without assertions are almost always pointless. You should at least assert the result of the query (empty, non-empty,...)